### ISSUE TITLE
Deleting a page in Craft does not invalidate the cache

### ISSUE THREAD

**Issue by vivienlacourba** [w3c/w3c-website#202]
On the staging site when I deleted a page from the Craft CMS I noticed that this page was still available several minutes later likely because the cache had not been purged for that specific URI. After some further time it finally returned a 404 meaning that the page got removed from the cache.

FYI the pages in question were:
- https://www-staging.w3.org/groups/how-we-work/
- https://www-staging.w3.org/cookies/

The excepted behavior is that similarly to editing a page in Craft which invalidates the cache, any page deletion should also invalidate the cache.

**Comment by JamieC24** [comment]
This issue appears to have been solved with the Craft 4 upgrade that was deployed in June.
Additionally, I have tested this on the dev environment and it seems to be solved.

**Comment by mnot** [comment]
Do you have the plugin installed and correctly configured? https://plugins.craftcms.com/cloudflare

**Comment by jean-gui** [comment]
@mnot I don't believe we use this plugin as we also rely on Cloudflare tags, but it could make sense to expand it rather than using our own. 

@JamieC24 If you tested on the restricted www-dev.w3.org, this server doesn't use caching at all. You would need to check if requests to CF are being issued by Craft when deleting. Seeing https://github.com/w3c/w3c-website-craft/blob/main/modules/cloudflare/Module.php#L83, it looks like the module only acts upon Entry::EVENT_AFTER_SAVE, not Entry::EVENT_AFTER_DELETE, which would explain the issue.

@simonrjones As you developed the module we use, what do you think about all this?

**Comment by JamieC24** [comment]
@jean-gui That would make sense! I'll have a look into that Cloudflare plugin for Craft - is there a way we could temporarily reinstate the cache on dev so I can test this?

**Comment by JamieC24** [comment]
@jean-gui I've made a [new branch for this] (https://github.com/w3c/w3c-website-craft/tree/fix/cloudflare-cache) - would it be possible to test this in an environment where the cloudflare cache is active?

**Comment by mnot** [comment]
Just to be clear - I have no experience with that plugin, so that's not an endorsement; I just did a search.

**Comment by JamieC24** [comment]
The Cloudflare Craft CMS plugin seems to work quite well for this. In terms of what caches it invalidates, it has the following options:


![Image](https://github.com/w3c/w3c-website/assets/122913995/deb6afa6-bee9-41df-a3a6-64ca555cccb9)

Having tested on dev with the cloudflare cache active, this seems to work quite well. Let me know if I'm clear to raise a pull request for this and we can see how it works on production

**Comment by simonrjones** [comment]
@jean-gui We've picked this up in a recent review and have updated the Craft module to fire after a delete as well as after a save. This is deployed to [CMS staging](https://cms-dev.w3.org/admin). However, I don't believe its possible to test this on staging (www-dev.w3.org) since Cloudflare doesn't cache these pages. 

See [PR #62 w3c/w3c-website-craft](https://github.com/w3c/w3c-website-craft/pull/62)

As a note, the [Cloudflare plugin](https://putyourlightson.com/plugins/cloudflare) doesn't clear cache by tags, which is why we have the custom module.
